tests/x11/yast2_snapper.pm

```diff
@@ -29,43 +29,87 @@ sub y2snapper_create_snapshot() {
     sleep 2;
     send_key "alt-o";
 }
+
+# Helper for selecting the new snapshot in y2-snapper
+#
+# Called when the list has been just loaded, so the top most item is selected
+sub y2snapper_select_snapshot() {
+    my $limit = 0; # Just in case the needles don't match at all (sh*t happens)
+
+    return 1 if (check_screen('yast2_snapper-new_snapshot_selected', 3));
+    # Return false if there is no snapshot to select
+    return 0 unless (check_screen('yast2_snapper-new_snapshot', 3));
+
+    until (check_screen('yast2_snapper-new_snapshot_selected', 5) || $limit > 20) {
+      $limit++;
+      send_key "down";
+    }
+    if (check_screen('yast2_snapper-new_snapshot_selected', 5)) {
+        return 1;
+    } else {
+        return 0;
+    }
+}
+
+# Quit yast2-snapper and cleanup the mess
+sub clean_and_quit() {
+    # C'l'ose  the snapper module
+    send_key "alt-l";
+    # Wait until xterm is focussed, delete the directory and close xterm
+    wait_idle;
+    script_run "rm -rf testdata";
+    script_run "ls";
+    script_run "exit";
+    save_screenshot;
+    script_run "exit";
+}
+
 sub run() {
     my $self = shift;
+
     # Make sure yast2-snapper is installed (if not: install it)
     ensure_installed "yast2-snapper";
+
     # Start an xterm as root
     x11_start_program("xterm");
     wait_idle;
     become_root;
     script_run "cd";
+
     # Start the yast2 snapper module and wait until it is started
     script_run "yast2 snapper";
-    assert_screen 'yast2_snapper-no_new_snapshot', 100;
+    assert_screen 'yast2_snapper-snapshots', 100;
+    # Make sure the test snapshot is not there
+    if (check_screen('yast2_snapper-new_snapshot', 5)) {
```

Comment:
maybe die "found a wrong snapshot exists" if check_screen('yast2_snapper-new_snapshot', 5);

at least we can the reason why it fail :)